### DISCUSSION TITLE
WebSockets in strange state after some time

### DISCUSSION THREAD

**Discussion by jorchg** [uNetworking/uWebSockets.js#167] [Q&A] [answered]
Hello @alexhultman 
I have been using your library for quite a few time. I was before using you `uws` library and recently moved to this new one.
We are facing a strange problem which is difficult to me to replicate because it involves sockets  opened for a long time.
We are using the library primarily for broadcasting data to connected clients. We use some sockets where the clients don't send any message in their entire lifecycle. We are also using long-lived connection for server-to-server communication. What we are seeing is that it seems that for these long-lived connections after some time, the sockets keep in a connected state but are not able to receive any data.
Now thinking of the past, maybe this was also present in your `uws` old library. We had a bug that we didn't thought it could be related to this.

I know the proper way to handle this is sending some `keep-alive` messages from clients to servers, but we do not want to overload our servers since this is a real-time app that needs to handle thousands of clients.

Any thoughts on this? Maybe setting a high `idleTimeout` could do the trick for us, we have some reconnection handling logic, but for me is kind of ugly hack.

Thank you very much. Congratulations on your superb work!

**Comment by ghost** [comment, marked as answer]
Hello Jorge,

Definitely use idleTimeout and let the clients send messages to the server from time to time. It should not be a problem for the server to receive. You don't even need to respond to them, just make sure to send something to the server.

With idle timeout of 120 seconds the server should pick up on and throw away dead connections properly

**Comment by slavaGanzin** [comment]
@jorchg 

Additionally, I recommend to force reconnection in case of failed keep-alive message and do not wait until user has intention to send you something. Show notification about closed connection to user. This leads to better user experience
